sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2808,6 +2808,53 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * (Scala-specific)
+   * Returns a new Dataset with a columns renamed.
+   * This is a no-op if schema doesn't contain existingName.
+   *
+   * `colsMap` is a map of existing column name and new column name.
+   *
+   * @group untypedrel
+   * @since 3.4.0
+   */
+  def withColumnsRenamed(colsMap: Map[String, String]): DataFrame = {
```

Comment:
What if there are duplicate new column names?

Reply:
Thats a good question and I thought about it. Before we discuss the approaches, should it be a responsibility of `withColumnsRenamed` to handle such a situation?